foyer/forcefield.py

```diff
@@ -298,7 +345,26 @@ def createSystem(self, topology, atomtype=True, nonbondedMethod=NoCutoff,
             the newly created System
         """
         if atomtype:
-            find_atomtypes(topology, forcefield=self)
+            independent_residues = _check_independent_residues(topology)
+
+            if independent_residues == True:
+                residue_map = dict()
+
+                for res in topology.residues():
+                    if res.name not in residue_map.keys():
+                        residue = _topology_from_residue(res)
+                        find_atomtypes(residue, forcefield=self)
+                        residue_map[res.name] = residue
+
+                new_topology = topology
+
+                for key, val in residue_map.items():
+                    new_topology = _update_atomtypes(topology, val)
+
+                topology = new_topology
```

Comment:
Again, this should be unnecessary if we operate directly on `topology` inside of `_update_atomtypes`